iree/hal/cuda/status_util.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/hal/cuda/status_util.h"
+
+#include "iree/hal/cuda/dynamic_symbols.h"
+
+iree_status_t iree_hal_cuda_result_to_status(
+    iree::hal::cuda::DynamicSymbols* syms, CUresult result, const char* file,
+    uint32_t line) {
+  if (result == CUDA_SUCCESS) {
+    return iree_ok_status();
+  }
+
+  const char* error_name;
```

Comment:
A good habit to get into is to always default-initialize locals - `const char* error_name = NULL;`. It's a defense-in-depth strategy: if the caller initializes, and the callee also initializes (by setting the output value to NULL/0/etc before doing any work), it's much less likely to end up being used uninitialized.